crates/typos-cli/src/file_type_specifics.rs

```diff
@@ -0,0 +1,76 @@
+//! This module specifies [`EngineConfig`] defaults for the file types defined in [`default_types`].
```

Comment:
I think it would make even more sense to move this module to the `typos-dict` crate ... however then we should also move the `default_types` module ... what do you think about that?

Reply:
For now, let's leave it here so we can get a better idea of usage.  For example, do we want per-format config besides dictionary information?

Reply:
(which the answer is already "yes" with `NO_CHECK_TYPES`)

Reply:
If/when we move some of this information, it wouldn't be in an array but in individual variables that we load up so nothing is tied to `default_types`

Reply:
I don't see any reason why we shouldn't move `default_types` to the `typos-dict`. But yeah I'm fine with leaving this here for now.